src/main/java/net/dv8tion/jda/internal/handle/GuildScheduledEventUpdateHandler.java

```diff
@@ -0,0 +1,131 @@
+        final GuildScheduledEvent.Status status = GuildScheduledEvent.Status.fromKey(content.getInt("status", -1));
+        final String imageUrl = content.getString("image", null);
+        String location = content.getString("channel_id", null);
+        GuildChannel channel = null;
+        if (location == null)
+            location = content.getObject("entity_metadata").getString("location");
+        else
+            channel = guild.getGuildChannelById(location);
+
+
+        if (!Objects.equals(name, event.getName()))
+        {
+            String oldName = event.getName();
+            event.setName(name);
+            getJDA().handleEvent(new GuildScheduledEventUpdateNameEvent(getJDA(), responseNumber, event, oldName));
+        }
+        if (!Objects.equals(description, event.getDescription()))
+        {
+            String oldDescription = event.getDescription();
+            event.setDescription(description);
+            getJDA().handleEvent(new GuildScheduledEventUpdateDescriptionEvent(getJDA(), responseNumber, event, oldDescription));
+        }
+        if (!Objects.equals(startTime, event.getStartTime()))
+        {
+            OffsetDateTime oldStartTime = event.getStartTime();
+            event.setStartTime(startTime);
+            getJDA().handleEvent(new GuildScheduledEventUpdateStartTimeEvent(getJDA(), responseNumber, event, oldStartTime));
+        }
+        if (!Objects.equals(endTime, event.getEndTime()))
+        {
+            OffsetDateTime oldEndTime = event.getEndTime();
+            event.setEndTime(endTime);
+            getJDA().handleEvent(new GuildScheduledEventUpdateEndTimeEvent(getJDA(), responseNumber, event, oldEndTime));
+        }
+        if (!Objects.equals(status, event.getStatus()))
+        {
+            GuildScheduledEvent.Status oldStatus = event.getStatus();
+            event.setStatus(status);
+            getJDA().handleEvent(new GuildScheduledEventUpdateStatusEvent(getJDA(), responseNumber, event, oldStatus));
+        }
+        if (channel == null && !Objects.equals(location, event.getExternalLocation()))
+        {
+            GuildScheduledEventUpdateLocationEvent.Location oldLocation = new GuildScheduledEventUpdateLocationEvent.Location(event);
+            event.setStageChannel(null);
+            event.setVoiceChannel(null);
+            event.setExternalLocation(location);
+            getJDA().handleEvent(new GuildScheduledEventUpdateLocationEvent(getJDA(), responseNumber, event, oldLocation));
+        }
+        if (channel instanceof StageChannel && !Objects.equals(channel, event.getStageChannel()))
+        {
+            GuildScheduledEventUpdateLocationEvent.Location oldLocation = new GuildScheduledEventUpdateLocationEvent.Location(event);
+            event.setVoiceChannel(null);
+            event.setExternalLocation(null);
+            event.setStageChannel((StageChannel) channel);
```

Comment:
I really dislike this pattern of having 3 values like this

Reply:
I removed this pattern of having 3 values and condensed it into just setLocation